app/src/main/java/com/ntg/user/mvpsample/add_edit_task/AddEditTaskFragment.java

```diff
@@ -0,0 +1,205 @@
+            } else {
+                presenter.saveTask(getNewTaskFromUser());
+            }
+        });
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_add_task, container, false);
+        ButterKnife.bind(this, view);
+        llm = new LinearLayoutManager(getContext());
+        subtasksRecyclerView.setLayoutManager(llm);
+        subTasksAdapter = new SubTasksAdapter(new ArrayList<>(0));
+        subtasksRecyclerView.setAdapter(subTasksAdapter);
+
+        return view;
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        View view = getLayoutInflater().inflate(R.layout.add_subtask_dialog, null);
+        switch (item.getItemId()) {
+            case R.id.action_add_subtask:
+                new AlertDialog.Builder(getContext())
+                        .setTitle("Add new SubTask")
+                        .setView(view)
+                        .setPositiveButton("Add", (dialogInterface, i) -> {
+                            EditText addSubTaskTitleEditText, addSubTaskProgressEditText;
+                            addSubTaskTitleEditText =
+                                    view.findViewById(R.id.et_add_subtask_title);
+                            addSubTaskProgressEditText =
+                                    view.findViewById(R.id.et_add_subtask_progress);
+                            String title = addSubTaskTitleEditText.getText().toString();
+                            int progress = Integer.parseInt(
+                                    addSubTaskProgressEditText.getText().toString());
+                            Subtask subtask = new Subtask(title, progress);
+                            subTasksAdapter.addSubTaskToList(subtask);
+
+                        })
+                        .setNegativeButton("Cancel",
+                                ((dialogInterface, i) -> dialogInterface.cancel()))
+                        .show();
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    @Override
+    public void setPresenter(AddEditTaskContract.Presenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void showErrorMsg(String errorMsg) {
+
+    }
+
+    /**
+     * This method return to tasks activity to show tasks after saving task operation
+     */
+    @Override
+    public void showTasks() {
+        getActivity().setResult(Activity.RESULT_OK);
+        getActivity().finish();
+    }
+
+    @Override
+    public void showTaskDetail(Task task) {
+        Intent intent = new Intent();
+        intent.putExtra("task", task);
```

Comment:
put bundle key as static variable in receiver side which expect this bundle (TaskDetailsActivity in this case)